sky/shell/platform/glfw/window_impl.cc

```diff
@@ -61,5 +71,66 @@ void WindowImpl::UpdateViewportMetrics(int width, int height) {
   engine_->OnViewportMetricsChanged(metrics.Pass());
 }
 
+void WindowImpl::DispatchMouseButtonEvent(int button, int action, int mods) {
+  pointer::PointerType type;
+  if (action == GLFW_PRESS) {
+    if (!buttons_) {
+      type = pointer::PointerType::DOWN;
+      glfwSetCursorPosCallback(window_, OnCursorPosChanged);
+    } else {
+      type = pointer::PointerType::MOVE;
+    }
+    buttons_ |= 1 << button;
+  } else if (action == GLFW_RELEASE) {
+    buttons_ &= !(1 << button);
+    if (!buttons_) {
+      type = pointer::PointerType::UP;
+      glfwSetCursorPosCallback(window_, nullptr);
+    } else {
+      type = pointer::PointerType::MOVE;
+    }
+  } else {
+    DLOG(INFO) << "Unknown mouse action: " << action;
+    return;
+  }
+
+  double x = 0.f, y = 0.f;
+  glfwGetCursorPos(window_, &x, &y);
+
+  base::TimeDelta time_stamp = base::TimeTicks::Now() - base::TimeTicks();
+
+  auto pointer_data = pointer::Pointer::New();
+  pointer_data->time_stamp = time_stamp.InMicroseconds();
+  pointer_data->type = type;
+  pointer_data->kind = pointer::PointerKind::MOUSE;
+  pointer_data->x = x;
+  pointer_data->y = y;
+  pointer_data->buttons = buttons_;
```

Comment:
If it makes you feel better, the value of `buttons` is dropped on the floor later in the pipeline.  https://github.com/flutter/flutter/issues/2899